src/Linux/Host/Broker/Impl/Services/LinuxSystemInfoService.cs

```diff
@@ -111,8 +118,103 @@ public class LinuxSystemInfoService : ISystemInfoService {
 
-        public (string VideoCardName, long VideoRAM, string VideoProcessor) GetVideoControllerInformation() {
-            return ("", 0, "");
+
+        private static object _videoCardInfoLock = new object();
+        private static List<VideoCardInfo> _videoCardInfo;
+        public IEnumerable<VideoCardInfo> GetVideoControllerInformation() {
+            lock (_videoCardInfoLock) {
+                if(_videoCardInfo != null) {
+                    return _videoCardInfo;
+                }
+
+                InitializeVideoCardInfo();
+                return _videoCardInfo;
+            }
+        }
+
+        private void InitializeVideoCardInfo() {
+            _videoCardInfo = new List<VideoCardInfo>();
+            const string vramPart = "VRAM:";
+            var lshwDisplayData = ExecuteAndGetOutput("/usr/bin/lshw", "-class display").Select(s => s.Trim()).ToArray();
+            var dmesgData = ExecuteAndGetOutput("dmesg", null).Where(s => s.ContainsIgnoreCase(vramPart)).ToArray();
+
+            string productPart = "product: ";
+            string vendorPart = "vendor: ";
+            string busPart = "bus info: ";
+
+            // Match anything that looks like 1024M or 1024MB
+            Regex pattern = new Regex("[0-9]+[GM][B]?", RegexOptions.IgnoreCase);
+
+            int count = 1;
+            for (int i=0;  i < lshwDisplayData.Length; ++i) {
+                // find the line that starts with "*-" 
+                if (lshwDisplayData[i].StartsWithIgnoreCase("*-")) {
+                    string productName = GetPart(lshwDisplayData, productPart, ref i);
+                    string vendorName = GetPart(lshwDisplayData, vendorPart, ref i);
+                    string busInfo = GetPart(lshwDisplayData, busPart, ref i);
+                    long vram = 0;
+                    if (!string.IsNullOrEmpty(busInfo)) {
+                        int index = busInfo.IndexOf('@');
+                        if (index >= 0 && index < busInfo.Length) {
+                            string busvalue = busInfo.Substring(index + 1);
+                            for (int j = 0; j < dmesgData.Length; ++j) {
+                                if (dmesgData[i].ContainsIgnoreCase(busvalue)) {
+                                    var match = pattern.Match(dmesgData[i]);
+                                    if (match.Success) {
+                                        vram = GetRamValueMB(match.Value);
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    _videoCardInfo.Add(new VideoCardInfo() { VideoCardName = productName, VideoRAM = vram, VideoProcessor = vendorName });
+                    ++count;
+                }
+            }
+        }
+
+        private static long GetRamValueMB(string vramStr) {
+            long ram = 0;
+            Regex numPattern = new Regex("[0-9]+");
```

Comment:
`new Regex("[0-9]+",  RegexOptions.Compiled);`, and let's make it a static field.